plugins/headlamp-plugin/config/webpack.config.js

```diff
@@ -1,3 +1,17 @@
+const externalModules = {
```

Comment:
The way this approach works is:
1. Headlamp's frontend already contains all the dependencies it needs for its UI, and plugins should use those too (not bundle them).
2. headlamp-plugin requires installing the dependencies useful for plugins, which should match the imports added to window.pluginLib and should match the versions in Headlamp's frontend. The reason why we add them here is to make them available for the plugin's editor to resolve the imports, e.g. for VS Code to find the modules and offer completion. These modules are not used to bundle the plugin code, i.e. they're not transpiled.
3. external modules is a way to tell *webpack* how to translate the imports. We do not want to bundle the dependencies code in the plugins, because that would risk using a different version of modules that already exist in Headlamp, and increase bundle size unnecessarily. So webpack translates e.g. `import {Button} from @material-ui/core` to `const { Button } = pluginLib.MuiCore`.

Conclusion: the way the code works should be exactly as if the user was still using `pluginLib`, the headlamp-plugin dependencies are just for helping with the coding experience.

Reply:
We could eventually make Headlamp itself look for dependencies from the plugins/headlamp-plugin . But let's think about that in the future.

Reply:
Thanks @joaquimrocha this does answer my question.